cmd/zed/agents/zfs_mod.c

```diff
@@ -445,16 +500,29 @@ zfs_iter_vdev(zpool_handle_t *zhp, nvlist_t *nvl, void *data)
 	} else if (dp->dd_compare != NULL) {
 		/*
 		 * NOTE: On Linux there is an event for partition, so unlike
-		 * illumos, substring matching is not required to accomodate
+		 * illumos, substring matching is not required to accommodate
 		 * the partition suffix. An exact match will be present in
 		 * the dp->dd_compare value.
 		 */
-		if (nvlist_lookup_string(nvl, dp->dd_prop, &path) != 0 ||
-		    strcmp(dp->dd_compare, path) != 0) {
+
+		if (nvlist_lookup_string(nvl, dp->dd_prop, &path) != 0)
 			return;
+		if (strcmp(dp->dd_compare, path) != 0) {
+			/*
+			 * Couldn't match the name, try stripping the
+			 * partition number off of it, and see if it
+			 * matches.
+			 */
+			spath = zfs_strip_partition_from_path(g_zfshdl, path);
```

Comment:
Not sure in which context this would be required. The devid key in the vdev label and in the disk event are generated by the same function.  And the physical path will never refer to a partition (slice).  So if this fixes something then I would suspect something else is broken.  TBD.